### ISSUE TITLE
Issue: dependency-cruise --init generates wrong regexes

### ISSUE THREAD

**Issue by mx-bernhard** [sverweij/dependency-cruiser#871]
The regexes generated by `dependency-cruise --init` seem to be wrong. The regex dot `.` matches more than the character `.` and should not be used to denote the start of the file extension part among other things.

### Expected Behavior

It should generate a config file that contains regexes for path/pathNot/etc. that match only the intended target files. In glob-speak these should match `*.test.ts` but not `*-test.ts` or `*-test-ts`.

### Current Behavior

E. g., `not-to-spec` has the following path set in my `.dependency-cruiser.cjs` after the `--init`:

```
path: '\.(spec|test)\.(js|mjs|cjs|ts|ls|coffee|litcoffee|coffee\.md)$',
```
Thus, it matches files like `use-mock-timers-for-test.ts`, `file-test-ts` (no extension) which it shouldn't since the usual pattern for tests is `.test.<extension>` with dots.

### Possible Solution

The generated Regex contains only one backslash instead of two. This is removed by prettier if configured due to the fact that a string in a .js-file like this `'foo\.bar'` is identical to `foo.bar`. When interpreted by regex means the dot-matcher instead of the actual dot.

### Steps to Reproduce (for bugs)

I used the regular `init` in a typescript project with `"type": "module"` enabled.

### Your Environment

- Version used: 15.3.0
- Node version: 18.18.2
- Operating System and version: WSL2 with Ubuntu 22.04

**Comment by mx-bernhard** [comment]
I always use `[.]` in my regexes instead of `\\.` to avoid escaping issues and I personally think it is more readable anyway.

**Comment by sverweij** [comment]
hi @mx-bernhard thanks for raising this bug! This slipped in when I kicked out handlebars templating ... . 

> I always use [.] in my regexes instead of \\. to avoid escaping issues and I personally think it is more readable anyway.

Good one! Hands done beats escaping escapes (as I'd needed to do in the  template: `\\\\\.`). 

Fix coming up.

**Comment by mx-bernhard** [comment]
You're welcome! Great to hear! 👍

**Comment by sverweij** [comment]
Hi again @mx-bernhard the fix has just been released as part of dependency-cruiser@15.4.0

**Comment by mx-bernhard** [comment]
I have since fixed them in our project, after having run `init`. So I was already good to go.
